src/main/flight/pid.c

```diff
@@ -89,26 +89,32 @@ FAST_DATA_ZERO_INIT float throttleBoost;
 pt1Filter_t throttleLpf;
 #endif
 
-PG_REGISTER_WITH_RESET_TEMPLATE(pidConfig_t, pidConfig, PG_PID_CONFIG, 3);
+PG_REGISTER_WITH_RESET_FN(pidConfig_t, pidConfig, PG_PID_CONFIG, 4);
 
+void pgResetFn_pidConfig(pidConfig_t *pidConfig)
+{
+    unsigned denom = 1;
 #if defined(STM32F411xE)
-#define PID_PROCESS_DENOM_DEFAULT       2
-#else
-#define PID_PROCESS_DENOM_DEFAULT       1
+    bool found = false;
+    for (unsigned pos = 0; pos < 7; pos++) {
+        if (gyroConfig()->gyrosDetected & BIT(pos)) {
+            found = true;
+            break;
+        }
+    }
+    if (found) {
+        if (gyro.gyroSensor1.gyroDev.gyroHardware != GYRO_BMI160 || gyro.gyroSensor1.gyroDev.gyroHardware != GYRO_BMI270) {
```

Comment:
Sensor1 and Sensor1?

Reply:
Could check witch gyro is active if not both and include in the check. Assuming most F411 boards won't have a second gyro.

Reply:
Note that you've repeated the same check twice...